File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {

        Arrays.sort(strs);  // flight , flow, flower
        StringBuilder sb= new StringBuilder(); 

        int n= strs.length; 

        String s1= strs[0];  //flight
        String s2= strs[n-1]; //flower
        int i=0,j=0; 

        while(i!=s1.length() && j!=s2.length()){
            if(s1.charAt(i)==s2.charAt(j)){
                sb.append(s1.charAt(i)); 
                i++; 
                j++; 
            }else {
                break; 
            }

        }
        return sb.toString(); 

        
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        prefix= strs[0] #dog
        
        plen=len(prefix)
        

        for i in range (1, len(strs)): 
            curr= strs[i] #racecar
            while prefix[:plen] != curr[:plen] or plen> len(curr)  : 
                plen -= 1
                if plen==0 : 
                    return ""
            prefix= prefix[:plen]
            #print(prefix)

        return prefix  
                



        